sentry/src/main/java/io/sentry/instrumentation/file/SentryFileInputStream.java

```diff
@@ -0,0 +1,144 @@
+    delegate = data.delegate;
+  }
+
+  private static FileInputStreamInitData init(
+      final @Nullable File file, @Nullable FileInputStream delegate, final @NotNull IHub hub)
+      throws FileNotFoundException {
+    final ISpan span = FileIOSpanManager.startSpan(hub, "file.read");
+    if (delegate == null) {
+      delegate = new FileInputStream(file);
+    }
+    return new FileInputStreamInitData(file, span, delegate, hub);
+  }
+
+  private static FileInputStreamInitData init(
+      final @NotNull FileDescriptor fd,
+      @Nullable FileInputStream delegate,
+      final @NotNull IHub hub) {
+    final ISpan span = FileIOSpanManager.startSpan(hub, "file.read");
+    if (delegate == null) {
+      delegate = new FileInputStream(fd);
+    }
+    return new FileInputStreamInitData(null, span, delegate, hub);
+  }
+
+  @Override
+  public int read() throws IOException {
+    // this is the only case, when the read() operation returns the byte value, and not the count
+    // hence we need this special handling
+    AtomicInteger result = new AtomicInteger(0);
+    spanManager.performIO(
+        () -> {
+          result.set(delegate.read());
+          return result.get() != -1 ? 1 : 0;
+        });
+    return result.get();
+  }
+
+  @Override
+  public int read(final byte @NotNull [] b) throws IOException {
+    return spanManager.performIO(() -> delegate.read(b));
+  }
+
+  @Override
+  public int read(final byte @NotNull [] b, final int off, final int len) throws IOException {
+    return spanManager.performIO(() -> delegate.read(b, off, len));
+  }
+
+  @Override
+  public long skip(final long n) throws IOException {
+    return spanManager.performIO(() -> delegate.skip(n));
+  }
+
+  @Override
+  public void close() throws IOException {
+    spanManager.finish(delegate);
+  }
+
+  public static final class Factory {
+    public static FileInputStream create(
+        final @NotNull FileInputStream delegate, final @Nullable String name)
+        throws FileNotFoundException {
+      return new SentryFileInputStream(
+          init(name != null ? new File(name) : null, delegate, HubAdapter.getInstance()));
+    }
+
+    public static FileInputStream create(
+        final @NotNull FileInputStream delegate, final @Nullable File file)
+        throws FileNotFoundException {
+      return new SentryFileInputStream(init(file, delegate, HubAdapter.getInstance()));
+    }
+
+    public static FileInputStream create(
+        final @NotNull FileInputStream delegate, final @NotNull FileDescriptor descriptor) {
+      return new SentryFileInputStream(
+          init(descriptor, delegate, HubAdapter.getInstance()), descriptor);
+    }
+
+    public static FileInputStream create(
+        final @NotNull FileInputStream delegate, final @Nullable File file, final @NotNull IHub hub)
+        throws FileNotFoundException {
+      return new SentryFileInputStream(init(file, delegate, hub));
+    }
```

Comment:
if this is for testing only, we can make package-private, or test the ctor directly

Reply:
If you mean Factory, nope, it's for the bytecode instrumentation, since we do wrapping there.

Reply:
Or if you mean this specific method, yep, it's only for testing

Reply:
this method specifically, its the only one that takes a `hub`